### ISSUE TITLE
import is failing

### ISSUE THREAD

**Issue by cesozgen** [pelias/vagrant#3]
I followed the instructions on this page (https://github.com/pelias/vagrant). However, the import of osm data is not working. I tried this multiple times (on linux and osx host) with different data sets.

==> default: [2015-01-16T07:31:18+00:00] INFO: remote_file[/opt/pelias/data/osm/san-francisco-bay_california.osm.pbf] sending run action to execute[load osm sfbay] (immediate)

==> default: 
==> default: ================================================================================
==> default: Error executing action `run` on resource 'execute[load osm sfbay]'
==> default: ================================================================================
==> default: 
==> default: 
==> default: Mixlib::ShellOut::CommandTimeout
==> default: --------------------------------
==> default: Command timed out after 14400s:
==> default: Command execeded allowed execution time, process terminated
==> default: ---- Begin output of node index.js >/opt/pelias/logs/osm_sfbay.out 2>/opt/pelias/logs/osm_sfbay.err ----
==> default: STDOUT: 
==> default: STDERR: 
==> default: ---- End output of node index.js >/opt/pelias/logs/osm_sfbay.out 2>/opt/pelias/logs/osm_sfbay.err ----
==> default: Ran node index.js >/opt/pelias/logs/osm_sfbay.out 2>/opt/pelias/logs/osm_sfbay.err returned 
==> default: 
==> default: 
==> default: Resource Declaration:
==> default: ---------------------
==> default: # In /tmp/vagrant-chef/023ea929f24e39a530508ed877b4cd02/cookbooks/pelias/recipes/osm.rb
==> default: 
==> default:  66:   execute "load osm #{name}" do
==> default:  67:     action  :nothing
==> default:  68:     user    node[:pelias][:user][:name]
==> default:  69:     command "node index.js >#{node[:pelias][:basedir]}/logs/osm_#{name}.out 2>#{node[:pelias][:basedir]}/logs/osm_#{name}.err"
==> default:  70:     cwd     "#{node[:pelias][:basedir]}/osm/current"
==> default:  71:     timeout node[:pelias][:osm][:timeout]
==> default:  72:     environment(
==> default:  73:       'HOME'                => node[:pelias][:user][:home],
==> default:  74:       'PELIAS_CONFIG'       => "#{node[:pelias][:cfg_dir]}/#{name}_#{node[:pelias][:cfg_file]}",
==> default:  75:       'OSMIUM_POOL_THREADS' => node[:pelias][:osm][:osmium_threads]
==> default:  76:     )
==> default:  77:   end
==> default:  78: end
==> default: 
==> default: 
==> default: 
==> default: Compiled Resource:
==> default: ------------------
==> default: # Declared in /tmp/vagrant-chef/023ea929f24e39a530508ed877b4cd02/cookbooks/pelias/recipes/osm.rb:66:in `block in from_file'
==> default: 
==> default: execute("load osm sfbay") do
==> default:   action [:nothing]
==> default:   retries 0
==> default:   retry_delay 2
==> default:   guard_interpreter :default
==> default:   command "node index.js >/opt/pelias/logs/osm_sfbay.out 2>/opt/pelias/logs/osm_sfbay.err"
==> default:   backup 5
==> default:   cwd "/opt/pelias/osm/current"
==> default:   environment {"HOME"=>"/home/pelias", "PELIAS_CONFIG"=>"/etc/pelias/sfbay_pelias.json", "OSMIUM_POOL_THREADS"=>"80"}
==> default:   returns 0
==> default:   timeout 14400
==> default:   user "pelias"
==> default:   cookbook_name :pelias
==> default:   recipe_name "osm"
==> default: end
==> default: 
==> default: 
==> default: 
==> default: [2015-01-16T12:24:42+00:00] INFO: Running queued delayed notifications before re-raising exception
==> default: [2015-01-16T12:24:42+00:00] INFO: ark[elasticsearch] sending start action to service[elasticsearch] (delayed)
==> default: [2015-01-16T12:24:43+00:00] INFO: git[/opt/pelias/pelias-api] sending restart action to runit_service[pelias-api] (delayed)
==> default: [2015-01-16T12:24:43+00:00] INFO: runit_service[pelias-api] restarted
==> default: [2015-01-16T12:24:43+00:00] ERROR: Running exception handlers
==> default: [2015-01-16T12:24:43+00:00] ERROR: Exception handlers complete
==> default: [2015-01-16T12:24:43+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
==> default: [2015-01-16T12:24:43+00:00] ERROR: execute[load osm sfbay] (pelias::osm line 66) had an error: Mixlib::ShellOut::CommandTimeout: Command timed out after 14400s:
==> default: Command execeded allowed execution time, process terminated
==> default: ---- Begin output of node index.js >/opt/pelias/logs/osm_sfbay.out 2>/opt/pelias/logs/osm_sfbay.err ----
==> default: STDOUT: 
==> default: STDERR: 
==> default: ---- End output of node index.js >/opt/pelias/logs/osm_sfbay.out 2>/opt/pelias/logs/osm_sfbay.err ----
==> default: Ran node index.js >/opt/pelias/logs/osm_sfbay.out 2>/opt/pelias/logs/osm_sfbay.err returned 
==> default: [2015-01-16T12:24:44+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.

------------------------------------------

The error file (/opt/pelias/logs/osm_sfbay.err) has many entries - I put a sample below.

...
{
  "osmium_mapper -> osm_types": 8000,
  "osm_types -> node_centroid_cache": 8000,
  "node_centroid_cache -> node_filter": 8000,
  "node_filter -> node_mapper": 4,
  "node_mapper -> node_hierarchyLookup": 4
}
{
  "osmium_mapper -> osm_types": 16000,
  "osm_types -> node_centroid_cache": 16000,
  "node_centroid_cache -> node_filter": 16000,
  "node_filter -> node_mapper": 4,
  "node_mapper -> node_hierarchyLookup": 4
}
...

**Comment by heffergm** [comment]
Could you paste me your pelias_settings.rb please? If it checks out, which I'm guessing it will, the issue is that the load of SF Bay data is taking longer than the allotted time given to that stage of the load process. You can increase it by overriding the following value in your config: node[:pelias][:osm][:timeout]. It defaults to 4 hours (14400 seconds).

**Comment by cesozgen** [comment]
I tried with different data sets - even with Nassau which has total 312KB data. So, I do not think timeout is the issue. Here is the file.

Vagrant.configure('2') do |config|
  config.vm.provision :chef_solo do |chef|
    chef.cookbooks_path = 'cookbooks'

    chef.json = {
      'pelias' => {
        'schema' => {
          'create_index' => true
        },
        'geonames' => {
          'index_data' => true,
          'alpha2_country_codes' => [
            'IT'
          ]
        },
        'quattroshapes' => {
          'index_data' => true,
          'alpha3_country_codes' => [
            'ITA'
          ]
        },
        'osm' => {
          'index_data' => true,
          'extracts' => {
            'sfbay'      => 'https://s3.amazonaws.com/metro-extracts.mapzen.com/san-francisco-bay_california.osm.pbf',
          }
        }
      }
    }

    chef.run_list = [
      'recipe[pelias::default]'
    ]
  end
end

**Comment by heffergm** [comment]
So for this to work you need to load both geonames and quattroshapes for the US, not for Italy (which are the defaults from the config).

I've not tried do so via the vagrant setup before, but expect it to take a fair bit of time. You'll need to drop your index (vagrant ssh, then curl -XDELETE http://localhost:9200/pelias) and update your config as follows:

```json
Vagrant.configure('2') do |config|
  config.vm.provision :chef_solo do |chef|
    chef.json = {
      'pelias' => {
        'api' => {
          'revision' => 'master'
        },
        'schema' => {
          'create_index' => true,
          'revision' => 'master'
        },
        'geonames' => {
          'index_data' => true,
          'revision' => 'master',
          'alpha2_country_codes' => [
            'US'
          ]
        },
        'quattroshapes' => {
          'index_data' => false,
          'revision' => 'master',
          'alpha3_country_codes' => [
            'USA'
          ]
        },
        'osm' => {
          'index_data' => true,
          'extracts' => {
            'sfbay' => 'https://s3.amazonaws.com/metro-extracts.mapzen.com/san-francisco-bay_california.osm.pbf',
          }
        }
      }
    }

    chef.run_list = [
      'recipe[pelias::default]'
    ]
  end
end
```

Then run `vagrant provision` to recreate the index and begin the load.

**Comment by cesozgen** [comment]
4 GB of memory is allocated BTW (I have not specified it but the script calculated the default as 4096).

**Comment by cesozgen** [comment]
I just saw your comment. I will make the modification that you suggested; I was not paying attention to that part.

Could you also send me the config for the world data? Do I need to list each country one by one? We are currently using Nominatim for reverse geocoding, but looking for alternatives because of its set of dependencies. Is Pelias a good alternative to Nominatim for world data? Do you have any guess about how long it will take to import world data?

Thank you for your help and quick response.

**Comment by heffergm** [comment]
Hi,
So to import US geonames/quattroshapes you're likely going to need a fair bit more memory. The vagrant environment was provided for local development on small datasets, as it's restricted to running, among other things, a single elasticsearch instance. You will not be able to load OSM planet inside a vagrant instance. To do so you'll need to build an Elasticsearch cluster. To give you a general idea of the size, we currently run 8x30GB nodes to support the planet. That is, however, with a replica, so for development you could get away with less than half that. Time to import the planet is around 3 weeks at present, although we're looking to shave a good bit of time off that in the near future.

**Comment by cesozgen** [comment]
I just tried with default settings (destroyed virtual machine, re-cloned the repo). The import is still running but I think it will timeout. And again error log files are not empty.

vagrant@pelias:/opt/pelias/logs$ ls -l
total 1340
-rw-r--r-- 1 pelias root     64 Jan 16 21:55 geonames_IT.err
-rw-r--r-- 1 pelias root  15741 Jan 16 21:56 geonames_IT.out
-rw-r--r-- 1 pelias root 422000 Jan 16 22:28 osm_rome.err
-rw-r--r-- 1 pelias root 156490 Jan 16 22:28 osm_rome.out
-rw-r--r-- 1 pelias root     74 Jan 16 21:57 quattroshapes_ITA_admin0.err
-rw-r--r-- 1 pelias root    452 Jan 16 21:57 quattroshapes_ITA_admin0.out
-rw-r--r-- 1 pelias root    302 Jan 16 21:57 quattroshapes_ITA_admin1.err
-rw-r--r-- 1 pelias root  12842 Jan 16 21:57 quattroshapes_ITA_admin1.out
-rw-r--r-- 1 pelias root     74 Jan 16 21:57 quattroshapes_ITA_admin2.err
-rw-r--r-- 1 pelias root    234 Jan 16 21:57 quattroshapes_ITA_admin2.out
-rw-r--r-- 1 pelias root    725 Jan 16 22:08 quattroshapes_ITA_local_admin.err
-rw-r--r-- 1 pelias root 309606 Jan 16 22:08 quattroshapes_ITA_local_admin.out
-rw-r--r-- 1 pelias root   2888 Jan 16 22:19 quattroshapes_ITA_locality.err
-rw-r--r-- 1 pelias root 370842 Jan 16 22:19 quattroshapes_ITA_locality.out
-rw-r--r-- 1 pelias root    194 Jan 16 22:19 quattroshapes_ITA_neighborhood.err
-rw-r--r-- 1 pelias root  15343 Jan 16 22:19 quattroshapes_ITA_neighborhood.out

-------

Sample error file: osm_rome.err

{
  "osmium_mapper -> osm_types": 6466,
  "osm_types -> node_centroid_cache": 6370,
  "node_centroid_cache -> node_filter": 6306,
  "node_filter -> node_mapper": 96,
  "node_mapper -> node_hierarchyLookup": 32
}
{
  "osmium_mapper -> osm_types": 6466,
  "osm_types -> node_centroid_cache": 6370,
  "node_centroid_cache -> node_filter": 6306,
  "node_filter -> node_mapper": 96,
  "node_mapper -> node_hierarchyLookup": 32
}
{
  "osmium_mapper -> osm_types": 6466,
  "osm_types -> node_centroid_cache": 6370,
  "node_centroid_cache -> node_filter": 6306,
  "node_filter -> node_mapper": 96,
  "node_mapper -> node_hierarchyLookup": 32,
  "node_hierarchyLookup -> node_meta.type": 3,
  "node_meta.type -> node_address_extractor": 3,
  "node_address_extractor -> node_suggester": 3,
  "node_suggester -> node_whitelist": 3,
  "node_blacklist -> es_node_dbclient_mapper": 3,
  "es_node_dbclient_mapper -> node_dbclient": 3
}
{
  "osmium_mapper -> osm_types": 6482,
  "osm_types -> node_centroid_cache": 6386,
  "node_centroid_cache -> node_filter": 6322,
  "node_filter -> node_mapper": 112,
  "node_mapper -> node_hierarchyLookup": 48,
  "node_hierarchyLookup -> node_meta.type": 17,
  "node_meta.type -> node_address_extractor": 17,
  "node_address_extractor -> node_suggester": 19,
  "node_suggester -> node_whitelist": 19,
  "node_blacklist -> es_node_dbclient_mapper": 19,
  "es_node_dbclient_mapper -> node_dbclient": 19
}
{
  "osmium_mapper -> osm_types": 6498,
  "osm_types -> node_centroid_cache": 6402,
  "node_centroid_cache -> node_filter": 6338,
  "node_filter -> node_mapper": 128,
  "node_mapper -> node_hierarchyLookup": 64,
  "node_hierarchyLookup -> node_meta.type": 35,
  "node_meta.type -> node_address_extractor": 35,
  "node_address_extractor -> node_suggester": 37,
  "node_suggester -> node_whitelist": 37,
  "node_blacklist -> es_node_dbclient_mapper": 37,
  "es_node_dbclient_mapper -> node_dbclient": 37
}
{
  "osmium_mapper -> osm_types": 6514,
  "osm_types -> node_centroid_cache": 6418,
  "node_centroid_cache -> node_filter": 6354,
  "node_filter -> node_mapper": 144,
  "node_mapper -> node_hierarchyLookup": 80,
  "node_hierarchyLookup -> node_meta.type": 55,
  "node_meta.type -> node_address_extractor": 55,
  "node_address_extractor -> node_suggester": 57,
  "node_suggester -> node_whitelist": 57,
  "node_blacklist -> es_node_dbclient_mapper": 57,
  "es_node_dbclient_mapper -> node_dbclient": 57
}
...

**Comment by heffergm** [comment]
Two things:

- the error log is not expected to be empty. You can follow progress by tailing the output (.out) log(s)
- there's a typo in the config I pasted, namely index_data should be set to 'true' for quattroshapes

**Comment by cesozgen** [comment]
>> the error log is not expected to be empty. You can follow progress by tailing the output (.out) log(s)

I would expect the log files provide some sort of information that the users can understand. In this case, it is not possible to understand from the error file whether the import is going well or not. I thought that since the file was not empty, something was wrong. For the last 35 minutes, there is no update in the osm_rome.out file, but osm_rome.err is growing (with similar information that I pasted above).

Also, I do not see any increase in elasticsearch index size.

vagrant@pelias:/opt/pelias/logs$ curl "http://localhost:9200/_cat/indices?v"
health index  pri rep docs.count docs.deleted store.size pri.store.size 
green  pelias   1   0     139210            0    483.5mb        483.5mb

>> Also, it appears to me that you're still trying to index Italy... I only see logs for IT/ITA and Rome osm output...

I am trying to do one successful import for a relatively small region, so I decided not to change the defaults that you guys provided.

I think it already passed one hour since I run vagrant up. VM is using 4 GB.

**Comment by heffergm** [comment]
It's most likely complete. There's a known bug with the osm import process not exiting on completion of the data load.

You can see how many records have been loaded:

```bash
types="geoname admin0 admin1 admin2 local_admin locality neighborhood osmnode osmway"
for i in $types; do
  echo "${i}:"
  curl http://localhost:9200/pelias/${i}/_count?pretty
done
```

**Comment by cesozgen** [comment]
But the process is still writing to error file.

Here is the output. How can I check these numbers from the original osm file?

geoname:
{
  "count" : 96628,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  }
}
admin0:
{
  "count" : 1,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  }
}
admin1:
{
  "count" : 103,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  }
}
admin2:
{
  "count" : 1,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  }
}
local_admin:
{
  "count" : 8081,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  }
}
locality:
{
  "count" : 8652,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  }
}
neighborhood:
{
  "count" : 377,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  }
}
osmnode:
{
  "count" : 19105,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  }
}
osmway:
{
  "count" : 6262,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  }
}

**Comment by heffergm** [comment]
It's done, it doesn't matter that it's still writing to the log: as I said, it's a known issue.

You can get counts from a pbf using osmconvert: osmconvert --out-statistics /tmp/auckland.osm.pbf

**Comment by cesozgen** [comment]
The imported data is much less than the original.

Here are the numbers from the original data.

vagrant@pelias:/tmp$ osmconvert --out-statistics rome_italy.osm.pbf 
lon min: 12.1090001
lon max: 12.8449984
lat min: 41.5780007
lat max: 42.1299986
nodes: 1344115
ways: 201277
relations: 10897
node id min: 14391521
node id max: 3269726371
way id min: 4079401
way id max: 320416175
relation id min: 5071
relation id max: 4456377
keyval pairs max: 263
keyval pairs max object: relation 365331
noderefs max: 1763
noderefs max object: way 224729171
relrefs max: 593
relrefs max object: relation 2759974

For example, nodes field here is 1344115, but I got 19105 from elasticsearch.

**Comment by heffergm** [comment]
As expected. A lot of what is in the export is thrown out during indexing.

**Comment by heffergm** [comment]
For reference, you can follow the bug opened against the openstreetmap import related to the process not exiting. In the meantime, know that it doesn't overtly affect things in the chef run, as the OSM import is the last step in the process. You can watch the .out logs for the relevant import and when the indexed document count stops incrementing, the process is done.

**Comment by cesozgen** [comment]
>> In the meantime, know that it doesn't overtly affect things in the chef run, as the OSM import is the last step in the process.

This is good to know; I was gonna ask this.

Thanks

**Comment by cesozgen** [comment]
Hi Grant,

Things are looking good (san francisco bay area import went fine). I will now try to import the planet data. I have a coupe of questions.

1. Where can I get a sample configuration file for importing world data? e.g. I do not want to list countries one by one for geonames and quattroshapes. Is there a shortcut for saying all countries?

2. Where should I ask this sort of questions? Do you guys have a group or should I just open as an issue here?

3. I would like to use this vm just as an import mechanism to another elastic search instance (which is running on another machine). Do you think this is a good idea? How can I configure the elastic search instance to be used (other than localhost)? Looks like https://github.com/pelias/config repo is not used for this vm.

Thanks

**Comment by heffergm** [comment]
1. geoname can be imported via `./bin/pelias-geonames -i all`. Quattroshapes is slightly more involved, but you can use the included runme.js and do the following:

```
types="admin0 admin1 admin2 local_admin locality neighborhood"
for i in ${types} do
  node example/runme.js ${type}
done
```

2. I'll get back to you on that one. For now this is fine.

3. Whether it's a good idea really depends on your environment. The vagrant environment does use pelias/config. The template for the configuration file can be found in cookbooks/pelias/templates/default/pelias.json (it installs in /etc/pelias/pelias.json). The elasticsearch connection information is hardcoded to localhost as this env wasn't built to facilitate the sort of thing you're looking to do, but you can of course build the environment and then override the configs. To reference the updated config if you run the import by hand, you can pass the path to the config file as an env var, e.g. PELIAS_CONFIG=/etc/pelias/pelias.json bin/pelias-geonames -i all, etc.

**Comment by cesozgen** [comment]
Hi Grant,

Thanks for earlier response. I have a few more questions.

1. I tried importing geoname data, quattroshapes data and osm data (provided by Mapzen at https://mapzen.com/metro-extracts), and everything seemed fine. Then I deleted the index, imported geoname data, skipped importing quattroshapes and tried to import some osm data (data for California fetched from http://download.geofabrik.de/north-america/us/california.html). Skipping the quattroshapes import did not work out well (it seems that that data is being used for osm import), I will make sure importing this data next time before importing osm data, but my question is not about this. The osm import took more than 2 days (and it did not finish, I killed the process). It only imported 455619 data points (this is the document count in elasticsearch). And I did not observe any CPU or memory bottleneck in either of the machines (the machine that imports the data and the machines that run the elasticsearch cluster). Is it possible that not importing quattroshapes data significantly slows down the osm import (does not seem very likely)? What can we do to speed up the imports? The extracts provided by Mapzen imports much faster, but you do not provide the whole world data, right? Which data source do you guys use to import the world data?

2. To cut down the import time for the world data, we are thinking to parallelize the process e.g. starting the import process for each country from a different client machine. Do you see any issues with this?

3. How do you guys handle the updates e.g. when osm data is updated? Do you have a way of doing incremental updates?

BTW, we are planning to use this just for reverse geocoding. Data correctness is important for us, but we do not need street level accuracy; we need city, state, country (and possibly point of interests).

Thanks a lot

**Comment by heffergm** [comment]
Hi,
With regards to:

1) as you discovered, importing OSM data without first importing geonames and quattroshapes won't work out well. I can't really speak to the state of an OSM import without first having done quattroshapes. 

As far as import time goes, I wouldn't expect California to take terribly long. The entire planet can be finished in roughly three weeks, and we're working on reducing that.

The planet data can be retrieved from any of the OSM mirror sites that host a full planet pbf. See http://wiki.openstreetmap.org/wiki/Planet.osm#Planet.osm_mirrors

2. I don't, if there were a requirement to import all the data this way, but really what you should do is import the planet as a single chunk. Download planet-latest.osm.pbf from one of the mirrors mentioned above.

3. We don't support incremental updates at present. It's in the works.

**Comment by cesozgen** [comment]
>> I don't, if there were a requirement to import all the data this way, but really what you should do is import the planet as a single chunk. Download planet-latest.osm.pbf from one of the mirrors mentioned above.

Why do you recommend importing the planet in a single chunk? Would not it be faster if for example import each continent in parallel from different client machines?

Thanks

**Comment by heffergm** [comment]
I suspect you'll hit the thresholds of performance for your cluster before such an exercise becomes worthwhile, but if you're up for trying it let us know how it goes.

**Comment by cesozgen** [comment]
Hi Grant,

Is there a way that we can get your elasticsearch snapshot (with the world data)? Importing osm data is really really slow. If you can provide us a snapshot, we can directly import that I think that will be just a few hours. Geonames and quattroshapes imports within a few hours, but osm import is slow (it will even take more than a month for us with this speed).

If that is not an option, I have another question. As I mentioned before, we just need high level information (city, state, country) from reverse geocoding. Do you think geonames + quattroshapes data will be enough for us regarding accuracy?

Thanks

**Comment by heffergm** [comment]
This is the most recent unadulterated geonames/quattroshapes/osm snapshot that I have: https://s3.amazonaws.com/mapzen.pelias-snapshots-1.3.4/

You're welcome to it for testing/etc. We'd discussed providing them before and decided against it for a variety of reasons, but let me know how it works out for you.

As for your question re: reverse geocoding, I'll enlist @missinglink to discuss.

**Comment by cesozgen** [comment]
Hi Grant,

Thanks a lot for the snapshot. I was able to restore it, and it works fine. We would like to use it for our production server (for reverse geocoding). Can we use it and what do we need to do to use it?

Thanks,
Cem

**Comment by heffergm** [comment]
You're welcome to use it as you like.  I'm not quite following your last question regarding what you need to do to use it.  Could you clarify?

**Comment by heffergm** [comment]
@cesozgen FYI, there's been a lot of recent work on the openstreetmap import pipeline. Right now that work is in the experimental branch (https://github.com/pelias/openstreetmap/tree/experimental) but it will likely be merged into master soon. Using the new pipeline, we've been doing full planet imports in ~6 days.

**Comment by cesozgen** [comment]
That is great news, thanks. The snapshot that you sent made things very easy for us. But I will give this a try when it is in master. Thanks for letting me know.

**Comment by cesozgen** [comment]
Hi Grant,

Is there a more recent snapshot than the one that you gave me last time? If yes, can I fetch it? For some of the osm points in New York area (11450 documents), state name is New Jersey. I was planning to write a script to fix those points. Maybe those are already fixed in the most recent snapshot.

Thanks in advance

**Comment by heffergm** [comment]
If by snapshot you mean extract, then those are updated weekly. They're built based on bounding boxes, not border tracing, so I'm sure there's a lot of NJ in the NY extract.

**Comment by cesozgen** [comment]
By snapshot, I mean elasticsearch snapshot. When I send a query something like
...
{ "match_phrase": {"admin1": "new jersey"}},
{ "match_phrase": {"admin2": "new york"}}
...
I get 11450 documents back.

The following is one of the reverse geo loookup results for World Trade Center

{"took":49,"timed_out":false,"_shards":{"total":20,"successful":20,"failed":0},"hits":{"total":957209,"max_score":null,"hits":[{"_index":"pelias_new","_type":"osmway","_id":"147229978","_score":null,"_source":{"center_point": {"lat": "40.713004", "lon": "-74.013186"}, "alpha3": "USA", "neighborhood": "Battery Park City", "name": {"bg": "\u0421\u0432\u0435\u0442\u043e\u0432\u0435\u043d \u0442\u044a\u0440\u0433\u043e\u0432\u0441\u043a\u0438 \u0446\u0435\u043d\u0442\u044a\u0440 1", "default": "One World Trade Center", "lt": "One World Trade Center", "ar": "\u0628\u0631\u062c \u0627\u0644\u062d\u0631\u064a\u0629", "uk": "\u0412\u0441\u0435\u0441\u0432\u0456\u0442\u043d\u0456\u0439 \u0442\u043e\u0440\u0433\u043e\u0432\u0438\u0439 \u0446\u0435\u043d\u0442\u0440 1", "pl": "1 World Trade Center", "he": "\u05de\u05e8\u05db\u05d6 \u05d4\u05e1\u05d7\u05e8 \u05d4\u05e2\u05d5\u05dc\u05de\u05d9 \u05d4\u05d7\u05d3\u05e9"}, "locality": "New York", "admin1": "New Jersey", "admin0": "United States", "admin2": "New York", "address": {"street": "World Trade Center", "number": "1", "zip": "10048"}, "local_admin": "Manhattan", "type": "way", "admin1_abbr": "NJ"},"sort":[0.09083542580464757]}]}}

**Comment by heffergm** [comment]
So you're looking for a full planet snapshot? We're not really planning to make those available. The build process for the planet is down to ~a day and a half now, which we feel is quite reasonable.

**Comment by cesozgen** [comment]
I see. A day and a half is pretty reasonable. I will give that a try. Thanks

**Comment by heffergm** [comment]
In fact, if you're only loading geonames/quattroshapes/osm, build time is probably around a day or less, although this is going to depend a lot on what your setup is like in terms of hardware, etc.

**Comment by cesozgen** [comment]
Hi Grant,

I tried the import with the current defaults (United Kingdom) with https://github.com/pelias/vagrant. I tried this import before doing the full planet import to test the latest changes. I am getting errors for the imports and the data in the elasticsearch is not complete. Is there a stable commit number that I can test the import? I tried twice (vagrant destroy and vagrant up) and got the same errors.

Here are the log files that contain errors:
https://www.dropbox.com/sh/9tamih0tr3fi181/AADXSwFtz4Uf75YjYQCHFAC4a?dl=0

Here is a sample query after import (see most of the fields are missing):
vagrant@pelias:~$ curl 'http://localhost:3100/reverse?lat=51.5181746&lon=-0.1106109'
{"type":"FeatureCollection","features":[{"type":"Feature","properties":{"id":"338463501","layer":"osmnode","name":"Barclays","text":"Barclays"},"geometry":{"type":"Point","coordinates":[-0.110725,51.518342]}}],"bbox":[-0.110725,51.518342,-0.110725,51.518342],"date":1430511077509}

**Comment by heffergm** [comment]
Master should be usable. What kind of hardware are you running this on? The finite strings errors can be ignored. The max_retries error at the end is indicative of you running low on resources and elasticsearch not being able to keep pace with the rate of the import.

**Comment by cesozgen** [comment]
Well, the machine has 32 GB Memory and 8 CPUs (Intel(R) Xeon(R) CPU E3-1230 V2 @ 3.30GHz) which has 4 cores in each. The limit would be whatever vagrant specifies.

**Comment by heffergm** [comment]
Can you confirm what revision you're on of the vagrant repo for me? I'm running an import now with all the defaults. The max_retries errors are actually normal now that I look at them, there are just some bad records in quattroshapes.

**Comment by cesozgen** [comment]
commit 88637923f1423b9682aa71d3a654e99788e49a31
Author: Grant Heffernan <heffergm@gmail.com>
Date:   Wed Apr 22 07:01:28 2015 -0400

    readme

**Comment by cesozgen** [comment]
But the resulting data is missing information. Compare the result that I gave above from what your api returns.

{"type":"FeatureCollection","features":[{"type":"Feature","properties":{"id":"338463501","layer":"osmnode","name":"Barclays","alpha3":"GBR","admin0":"United Kingdom","admin1":"Camden","admin2":"Greater London","locality":"London","neighborhood":"Lincolns Inn","text":"Barclays, Lincolns Inn, Greater London"},"geometry":{"type":"Point","coordinates":[-0.110725,51.518342]}}],"bbox":[-0.110725,51.518342,-0.110725,51.518342],"date":1430513093994}

Also osm import did not seem to complete succesfully, there are only 50k points which seem low to me.

geoname:{"count":57531,"_shards":{"total":1,"successful":1,"failed":0}}
admin0:{"count":1,"_shards":{"total":1,"successful":1,"failed":0}}
admin1:{"count":232,"_shards":{"total":1,"successful":1,"failed":0}}
admin2:{"count":0,"_shards":{"total":1,"successful":1,"failed":0}}
local_admin:{"count":11694,"_shards":{"total":1,"successful":1,"failed":0}}
locality:{"count":1770,"_shards":{"total":1,"successful":1,"failed":0}}
neighborhood:{"count":2350,"_shards":{"total":1,"successful":1,"failed":0}}
osmnode:{"count":52405,"_shards":{"total":1,"successful":1,"failed":0}}
osmway:{"count":24,"_shards":{"total":1,"successful":1,"failed":0}}

**Comment by cesozgen** [comment]
There is no problem with geoname data. The result that you gave is from the geoname data. I am having problem with osm data.

**Comment by heffergm** [comment]
Your data looks correct to me, and had any of the imports actually failed, the chef run would have exited. For OSM you forgot to include the osmaddress type in your count, which is another 28k records.

The results you get in the default vagrant setup won't be the same as our API. Among other things, we do admin heirarchy lookups against quattroshapes in memory, which isn't feasible for the vast majority of people using vagrant. The footprint generally just isn't big enough to support doing so as you'll need something north of 15GB of available RAM (8gb to load quattro into memory and the remainder to deal with everything else assuming you're running all this on one node).

If you want to try it, you can:
- up the available RAM allocated to vagrant in Vagrantfile (or via the PELIAS_VAGRANT_MB environment variable)
- set the following keys to true in your pelias settings file:

[:pelias][:geonames][:admin_lookup]
[:pelias][:quattroshapes][:admin_lookup]
[:pelias][:osm][:admin_lookup]

**Comment by cesozgen** [comment]
Thanks, it is working fine now, I believe you guys recently added these options.

**Comment by cesozgen** [comment]
Also, you wrote in your comment "Among other things". What other tweaks do I need in order to replicate your production data (if you guys are willing to provide :))?

**Comment by heffergm** [comment]
Just a figure of speech. I think you've got everything.

**Comment by heffergm** [comment]
Actually, I just realized that isn’t entirely the case. We do load Openaddresses data in our environment, which isn’t part of the vagrant setup. Vagrant was really just to give people an overview of the basics and how to get them working, so building in importers for every dataset we have wasn’t at the top of the list. If you’re interested I can either give you the basics or maybe work on adding it to vagrant time permitting.

--
Grant Heffernan

> Il giorno 01/mag/2015, alle ore 08:30 PM, Grant Heffernan <heffergm@gmail.com> ha scritto:
> 
> Just a figure of speech. I think you've got everything.
> 
>

**Comment by cesozgen** [comment]
Hi Grant,

There are many unicode related errors in the database. Are you guys planning to fix those? Any suggestions on how to deal with them? I searched the character º in the locality field only (which is not a valid character) and the search yielded more than 1000 unique results.

http://pelias.mapzen.com/reverse?lat=21.090333&lon=-86.769797
"locality": "CancÃºn",

http://pelias.mapzen.com/reverse?lat=45.519000&lon=-73.584000
"locality": "MontrÃ©al"

Thx,
Cem

**Comment by abuzakaria** [comment]
Hi guys. Can you update the snapshot link? Would be better if an updated one. :smile:
